### PR TITLE
[OSD-21571] Create secret to configure OBO Alertmanager

### PR DESCRIPTION
**Pull request by xiaoyu74**
### What type of PR is this?
_(feature)_

### What this PR does / why we need it?
Create secret to configure OBO Alertmanager across fleet of HCP Management Clusters to route alerts payload to ocm-agent-fleet

### Which Jira/Github issue(s) this PR fixes?
https://issues.redhat.com/browse/OSD-21571

### CODE REVIEW

File: deploy/hypershift-obo-alertmanager-config/config.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+deploymentMode: "SelectorSyncSet"
+SelectorSyncSet:
+  resourceApplyMode: Sync
+  matchExpressions:
```
Comment (bmeng) [inline deploy/hypershift-obo-alertmanager-config/config.yaml:4]: I think we need to deploy this on management cluster only, you can refer to the other one for ocm-agent  SSS.

File: deploy/hypershift-obo-alertmanager-config/hypershift-obo-alertmanager-config.secret.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: v1
+data:
+  alertmanager.yaml: Z2xvYmFsOgogIHJlc29sdmVfdGltZW91dDogNW0Kcm91dGU6CiAgcmVjZWl2ZXI6ICJudWxsIgogIHJvdXRlczoKICAtIHJlY2VpdmVyOiBvY21hZ2VudAogICAgbWF0Y2hlcnM6CiAgICAtIHNlbmRfbWFuYWdlZF9ub3RpZmljYXRpb24gPSB0cnVlCiAgICByZXBlYXRfaW50ZXJ2YWw6IDEwbQogIC0gcmVjZWl2ZXI6ICJudWxsIgogIGdyb3VwX3dhaXQ6IDMwcwogIGdyb3VwX2ludGVydmFsOiA1bQogIHJlcGVhdF9pbnRlcnZhbDogMTJoCnJlY2VpdmVyczoKLSBuYW1lOiBvY21hZ2VudAogIHdlYmhvb2tfY29uZmlnczoKICAtIHNlbmRfcmVzb2x2ZWQ6IHRydWUKICAgIHVybDogaHR0cDovL29jbS1hZ2VudC1mbGVldC5vcGVuc2hpZnQtb2NtLWFnZW50LW9wZXJhdG9yLnN2Yy5jbHVzdGVyLmxvY2FsOjgwODEvYWxlcnRtYW5hZ2VyLXJlY2VpdmVyCi0gbmFtZTogIm51bGwiCg==
```
Comment (bmeng) [inline deploy/hypershift-obo-alertmanager-config/hypershift-obo-alertmanager-config.secret.yaml:3]: url: http://ocm-agent-fleet.openshift-ocm-agent-operator.svc.cluster.local:8081/alertmanager-receiver

not sure if we can use a variable here

File: deploy/hypershift-obo-alertmanager-config/config.yaml (hunk 1 above)
Comment (xiaoyu74) [reply deploy/hypershift-obo-alertmanager-config/config.yaml:4]: @bmeng - You mean only keep the key section below? 
```
key: api.openshift.com/managed
```
I saw both `ocm-agent-operator-managedfleetnotifications` and `ocm-agent-operator-managednotifications` have SSS with the `fedramp` only which indicates U.S. government-wide clusters?
```
- key: api.openshift.com/fedramp                                                                
```

Comment (bmeng) [reply deploy/hypershift-obo-alertmanager-config/config.yaml:4]: to deploy on management cluster only, we need https://github.com/openshift/ocm-agent/blob/master/hack/00-ocm-agent.selectorsyncset.yaml.tmpl#L49

File: deploy/hypershift-obo-alertmanager-config/hypershift-obo-alertmanager-config.secret.yaml (hunk 2 above)
Comment (xiaoyu74) [reply deploy/hypershift-obo-alertmanager-config/hypershift-obo-alertmanager-config.secret.yaml:3]: > url: http://ocm-agent-fleet.openshift-ocm-agent-operator.svc.cluster.local:8081/alertmanager-receiver
> 
> not sure if we can use a variable here

You mean having the variable store in a separated file in the same folder of the secret file and just defined the variable name in the secret.yaml?  Let me check if there is any existing SSS following the pattern.

Comment (bmeng) [reply deploy/hypershift-obo-alertmanager-config/hypershift-obo-alertmanager-config.secret.yaml:3]: hmm, never mind, I think we hard coded the ocm-agent name here as well. https://github.com/openshift/ocm-agent/blob/master/hack/00-ocm-agent.selectorsyncset.yaml.tmpl#L55
So it should be ok.

**Pull request merged**